### PR TITLE
CHANGELOG.md with Markdown links

### PR DESCRIPTION
**Pull request by olleolleolle**
This adds clickable links to PRs and Issues for the part of the CHANGELOG document where those point to this GitHub.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,102 +1,102 @@
 ## 2.0.1 / 2018-02-17
 
-* Repair nested namespaces, by avoiding prefix duplication #1322. Fixes #1310 by Kunpei Sakai
+* Repair nested namespaces, by avoiding prefix duplication [#1322](https://github.com/sinatra/sinatra/issues/1322). Fixes [#1310](https://github.com/sinatra/sinatra/issues/1310) by Kunpei Sakai
 
-* Add pattern matches to values for Mustermann::Concat #1333. Fixes #1332 by Dawa Ometto
+* Add pattern matches to values for Mustermann::Concat [#1333](https://github.com/sinatra/sinatra/issues/1333). Fixes [#1332](https://github.com/sinatra/sinatra/issues/1332) by Dawa Ometto
 
-* Ship the VERSION file with the gem, to allow local unpacking #1338 by Olle Jonsson
+* Ship the VERSION file with the gem, to allow local unpacking [#1338](https://github.com/sinatra/sinatra/issues/1338) by Olle Jonsson
```
Comment (namusyaka) [inline CHANGELOG.md]: I think we could also link user names, for example:

`[Olle Jonsson](https://github.com/olleolleolle)`

Comment (olleolleolle) [reply CHANGELOG.md]: @namusyaka If we have the user names, it's easy. I'd say, finding them now is "hard".

Comment (olleolleolle) [reply CHANGELOG.md]: @namusyaka We could incrementally build up a list of linked names?

Comment (namusyaka) [reply CHANGELOG.md]: Okay, no worries. Let's think after releasing v2.0.2!

**Pull request merged**